### PR TITLE
Update: React hook form

### PR DESCRIPTION
**Pull request by maxplt**
# Jira Issue: []

## What?
Update `react-hook-form` to the latest version.
Also, tried to use `react-hook-form` in another way. 

## Why?
Because we killed the `Semantic-UI`, we don't need to use unnecessary `Controllers` on every component. 

## Optional checklist
- [ ] Updated `src/changelog.ts`
- [ ] Codescouted
- [ ] Unit tests written
- [ ] Tested locally

### CODE REVIEW

File: shared/@hedvig-ui/Dropdown/dropdown.tsx [hunk 1]
```diff
@@ -290,3 +290,27 @@ export const Option: React.FC<OptionProps> = ({
     {children}
   </OptionStyled>
 )
+
+// TODO: More semantic way?
```
Comment (vonElfvin) [inline shared/@hedvig-ui/Dropdown/dropdown.tsx:294]: Would this make it possible to not send a control down?

File: src/portals/hope/features/claims/claim-details/ClaimPayments/components/ClaimPaymentForm.tsx [hunk 2]
```diff
@@ -170,212 +164,146 @@ export const ClaimPaymentForm: React.FC<{
...
+        style={{ width: '8rem', marginBottom: '1.5rem' }}
+        checked={exGratia}
+        onChange={() => setExGratia((prev) => !prev)}
+      />
 
-        <FormDropdown
-          placeholder="Type"
-          options={categoryOptions.filter((opt) => {
+      {/* In that case, we need to use Controller,
+          because Dropdown is an unusual field. */}
```
Comment (vonElfvin) [inline src/portals/hope/features/claims/claim-details/ClaimPayments/components/ClaimPaymentForm.tsx:224]: How come? :)

File: src/portals/hope/features/claims/claim-details/ClaimPayments/components/ClaimPaymentForm.tsx [hunk 3]
```diff
@@ -170,212 +164,146 @@ export const ClaimPaymentForm: React.FC<{
...
+      />
+      <Input
+        {...register('deductible')}
+        placeholder="Deductible"
+        step="any"
+        type="number"
+        style={{ marginBottom: '1.5rem' }}
+      />
+      <Input
+        {...register('note', { required: true })}
```
Comment (vonElfvin) [inline src/portals/hope/features/claims/claim-details/ClaimPayments/components/ClaimPaymentForm.tsx:209]: Any way to specify the feedback? "Note is required" to the left

File: src/portals/hope/features/claims/claim-details/ClaimPayments/components/ClaimPaymentForm.tsx [hunk 4]
```diff
@@ -170,212 +164,146 @@ export const ClaimPaymentForm: React.FC<{
...
+        placeholder="Note"
+        error={errors.note?.type === 'required'}
+        errorText="Note is required"
```
Comment (vonElfvin) [inline src/portals/hope/features/claims/claim-details/ClaimPayments/components/ClaimPaymentForm.tsx:212]: Feels like this logic could live inside the component somehow

Comment (vonElfvin) [reply src/portals/hope/features/claims/claim-details/ClaimPayments/components/ClaimPaymentForm.tsx:212]: Feels like the "developer UX" was slightly neater before

**Review by vonElfvin — COMMENTED** [review]
Cool! Added some questions :)

File: src/portals/hope/features/claims/claim-details/ClaimPayments/components/PaymentConfirmationModal.tsx [hunk 5]
```diff
@@ -39,37 +146,29 @@ export const PaymentConfirmationModal: React.FC<
...
-            onClose()
-          }}
-        >
-          Cancel
-        </Button>
-      </ButtonsGroup>
+      <form onSubmit={handleSubmit(submitHandler)}>
+        <Input
+          autoFocus
+          {...register('confirmation', { required: true })}
```
Comment (rasmusguterstam) [inline src/portals/hope/features/claims/claim-details/ClaimPayments/components/PaymentConfirmationModal.tsx:152]: You see where I got the inspiration to `useNavigation` from 🤭

**Review by rasmusguterstam — COMMENTED** [review]
Not really sure what state this PR is in, still a draft? Looks good though, so let me know when everything works as expected and you want a proper review! 🤝

File: shared/@hedvig-ui/Dropdown/dropdown.tsx (hunk 1 above)
Comment (maxplt) [reply shared/@hedvig-ui/Dropdown/dropdown.tsx]: Yep. I created this one in new form component

**Review by rasmusguterstam — COMMENTED** [review]
I don't really have any specific comments on the code, but some general thoughts.

Do we really need all these form wrappers now that we have the new version? An example is the `Input` component, which has its equivalent of `FormInput` which basically wraps it with a `FormField`. Why don't we just make sure that every relevant component in `@hedvig-ui` can spread `register` and render accordingly? Wouldn't it be sweet if the `Input` component in general has "label" and "error" as a prop, to be used even if it wasn't a part of a form?

File: shared/@hedvig-ui/Input/input.tsx [hunk 6]
```diff
@@ -188,45 +203,61 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>(
...
+          {isSuccess && <SuccessIcon affix={isAffix ? 1 : 0} size={size} />}
+          {isError && <ErrorIcon affix={isAffix ? 1 : 0} size={size} />}
+        </InputWrapper>
+        {name && errors && (
+          <ErrorMessage
+            errors={errors}
+            name={name}
+            render={({ message }) => <Error>{message}</Error>}
+          />
+        )}
```
Comment (rasmusguterstam) [inline shared/@hedvig-ui/Input/input.tsx:259]: Very nice!

File: src/portals/hope/features/claims/claim-details/ClaimPayments/components/ClaimPaymentForm.tsx [hunk 7]
```diff
@@ -156,237 +140,142 @@ export const ClaimPaymentForm: React.FC<{
...
+    <Form onSubmit={handleSubmit(submitHandler)}>
+      <Input
+        {...register('amount', { required: 'Amount is required' })}
+        placeholder="Payout amount"
+        type="number"
+        step="any"
+        affix={{ content: 'SEK' }}
+        errors={errors}
+        label="Amount"
+      />
```
Comment (rasmusguterstam) [inline src/portals/hope/features/claims/claim-details/ClaimPayments/components/ClaimPaymentForm.tsx:174]: Haha woow so much better

**Review by rasmusguterstam — APPROVED** [review]
🙌  (just some tests to fix 🤓 )

**Pull request merged**